src/Authentication/Helpers/SpecificDecisionHelper.cs

```diff
@@ -0,0 +1,371 @@
+    private const string DefaultType = "string";
+    private const string PersonHeaderTrigger = "person";
+    private const string OrganizationHeaderTrigger = "organization";
+    private const string PersonHeader = "Altinn-Party-SocialSecurityNumber";
+    private const string OrganizationNumberHeader = "Altinn-Party-OrganizationNumber";
+    private const string PolicyObligationMinAuthnLevelOrg = "urn:altinn:minimum-authenticationlevel-org";
+    private const string PolicyObligationMinAuthnLevel = "urn:altinn:minimum-authenticationlevel";
+
+    /// <summary>
+    /// SystemUser id
+    /// Should be exported to the AltinnCoreClaimTypes class
+    /// </summary>
+    public const string SystemUserUuid = "urn:altinn:systemuser:uuid";
+
+    /// <summary>
+    /// Resource Registry attribute match indentifier 
+    /// Should be exported to the AltinnCoreClaimTypes class
+    /// </summary>
+    public const string ResourceRegistryAttribute = "urn:altinn:resource";
+
+    /// <summary>
+    /// Attribute Matching Identity.
+    /// Should be exported to the AltinnCoreClaimTypes class
+    /// </summary>
+    public const string ActionId = "urn:oasis:names:tc:xacml:1.0:action:action-id";
+
+    private static readonly JsonSerializerOptions Options = new() { PropertyNameCaseInsensitive = true };
+
+    /// <summary>
+    /// Creates a decision request based on input
+    /// </summary>
+    /// <returns></returns>
+    public static XacmlJsonRequestRoot CreateDecisionRequest(AuthorizationHandlerContext context, EndUserResourceAccessRequirement requirement, IQueryCollection queryParams)
+    {
+        XacmlJsonRequest request = new XacmlJsonRequest();
+        request.AccessSubject = new List<XacmlJsonCategory>();
+        request.Action = new List<XacmlJsonCategory>();
+        request.Resource = new List<XacmlJsonCategory>();
+
+        string party = queryParams.FirstOrDefault(p => p.Key == ParamParty).Value.FirstOrDefault();
+
+        request.AccessSubject.Add(CreateSubjectCategory(context.User.Claims));
+        request.Action.Add(CreateActionCategory(requirement.ActionType));
+
+        XacmlJsonCategory resource = CreateResourceCategoryForResource(requirement.ResourceId);
+        request.Resource.Add(resource);
+
+        if (Guid.TryParse(party, out Guid partyUuid))
+        {
+            resource.Attribute.Add(CreateXacmlJsonAttribute(AltinnCoreClaimTypes.PartyUUID, partyUuid.ToString(), DefaultType, DefaultIssuer));
+        }
+        else
+        {
+            throw new ArgumentException("invalid party " + party);
+        }
+
+        XacmlJsonRequestRoot jsonRequest = new() { Request = request };
+
+        return jsonRequest;
+    }
+
+    /// <summary>
+    /// Creates a decision request based on input
+    /// </summary>
+    /// <returns></returns>
+    public static XacmlJsonRequestRoot CreateDecisionRequestManual(IEnumerable<Claim> claims, EndUserResourceAccessRequirement requirement, Guid partyUuid)
```

Comment:
Litt usikker på bruken av EndUserResourceAccessRequirement.  Den er relevant for handlere og har brukes den bare som en DTO for action og ressursid.  Hadde det ikke vært like enkelt bare å ha resurce id og action som input params til metoden